File: app/src/main/java/com/example/jetpackcomponentsapp/view/holder/CustomViewHolder.java
package com.example.jetpackcomponentsapp.view.holder;

import android.view.View;

import com.example.jetpackcomponentsapp.databinding.CustomBinder;
import com.example.jetpackcomponentsapp.model.CustomModel;
import com.example.jetpackcomponentsapp.view.CustomListeners;

public class CustomViewHolder extends BaseViewHolder {

    private final CustomBinder binder;

    public CustomViewHolder(CustomListeners listeners, CustomBinder binder ) {
        super(listeners,binder .getRoot());
        this.binder = binder;
    }

    @Override
    public void bindDataToViewHolder(final CustomModel model, final int position) {
        binder.setCustomModel(model);
        binder.imageView.setImageResource(model.getIcon());
        binder.buttonEdit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                getListener().onUpdate(model,position);
            }
        });
        binder.buttonDelete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                getListener().onDelete(model,position);
            }
        });
    }
}

File: app/src/main/java/com/example/jetpackcomponentsapp/view/holder/CustomViewHolder.kt
package com.example.jetpackcomponentsapp.view.holder

import android.view.View
import com.example.jetpackcomponentsapp.view.CustomListeners
import com.example.jetpackcomponentsapp.model.CustomModel
import com.example.jetpackcomponentsapp.databinding.CustomBinder

class CustomViewHolder : BaseViewHolder {

    private var customBinder : CustomBinder

    constructor(customListeners : CustomListeners, customBinder : CustomBinder) : super(customListeners, customBinder.root) {
        this.customBinder = customBinder
    }

    override fun bindDataToViewHolder(model : CustomModel, position : Int) {
        customBinder.setCustomModel(model)
        customBinder.executePendingBindings()
        //customBinder.imageView.setBackgroundResource(item.icon ?: 0)
        customBinder.imageView.setImageResource(model.icon)
        customBinder.buttonEdit.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view : View) {
                getListener().onUpdate(model ,position)
            }
        })
        customBinder.buttonDelete.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view : View) {
                getListener().onDelete(model, position)
            }
        })
    }
}